Frame.cc

```diff
@@ -889,12 +889,34 @@ bool Frame::FillSpectralProfileData(
         if (!region->IsValid()) {
             return false;
         }
+
+        int curr_stokes(CurrentStokes());
+
+        const casacore::ArrayLattice<casacore::Bool>* mask = nullptr;
+        std::unique_lock<std::mutex> guard(_image_mutex);
+        try {
+            // check is the region mask valid (outside the lattice or not)
+            mask = region->XyMask();
+        } catch (casacore::AipsError& err) {
+        }
+        guard.unlock();
+        if (!mask) {
```

Comment:
When SET_IMAGE_CHANNELS is used to change the stokes, new spectral profile data is sent for all regions which have requirements set (checked below in line 915).  This is sending a NaN profile before this check, for regions which may not require spectral profiles.

Reply:
Good point. I moved the check closer to where it had been before, since there are two other checks which should also be done beforehand.